### ISSUE TITLE
Implementation of ol3 groups

### ISSUE THREAD

**Issue by lewisjared** [tombatossals/angular-openlayers-directive#179]
I am looking to implement groups which will help solve #59. I see two obvious solutions that groups can be added and I was looking for some feedback before I get going.

1. Add ol-group and use dom hierarchy to show relationships
```
<ol-layer ol-layer-properties="someLayer></ol-layer>
<ol-group>
    <ol-layer ol-layer-properties="layer" ng-repeat="layer in otherLayers"></ol-layer>
</ol-group
```

2. Add group layer type in olHelpers.createLayer. For a group layer, the properties object has a layers attribute. An example of the configuration is shown below:
```
{
    name: 'GeoJSON',
    type: 'Group',
    layers: [
        {
            name: 'Italy',
            source: {
                type: 'GeoJSON',
                url: 'json/ITA.geo.json'
            }
        },
        {
            name: 'France',
            source: {
                type: 'GeoJSON',
                url: 'json/FRA.geo.json'
            },
        },
        {
            name: 'Spain',
            source: {
                type: 'GeoJSON',
                url: 'json/ESP.geo.json'
            }
        },
        {
            name: 'Lisbon',
            source: {
                type: 'GeoJSON',
                url: 'json/LIS.geo.json'
            }
        } 
    ]
}
```

1 is more angular, but will be much more difficult to implement. I had a working implementation of 2 on an older version of the library. Another option is both and let users choose. Any suggestions/feedback?

Edited: Removed top level attribute 'source' in sample layer object

**Comment by claustres** [comment]
IMHO we need something between both approaches. The first one is not really appropriate if you want to manage layer grouping (add/remove to/from a group) dynamically from the controller code because you will have to manipulate the DOM. The second one is maybe better but you have no option to move layers between groups except by recreating everything.

So I would prefer the layer directive allowing to declare a group type layer like `<ol-layer ng-repeat="group in groups" ol-layer-properties="group"><ol-layer/>` then an attribute on the others layers to indicate if they are in a group or not like `<ol-layer ng-repeat="layer in layers" ol-layer-properties="layer"></ol-layer>` with this scope:
```javascript
...
group: [{
    name: 'Countries',
    type: 'Group'
  },
  {
    name: 'Cities',
    type: 'Group'
  }],
layers: [
        {
            name: 'Italy',
            group: 'Countries',
            source: {
                type: 'GeoJSON',
                url: 'json/ITA.geo.json'
            }
        },
        {
            name: 'France',
            group: 'Countries',
            source: {
                type: 'GeoJSON',
                url: 'json/FRA.geo.json'
            },
        },
        {
            name: 'Spain',
            group: 'Countries',
            source: {
                type: 'GeoJSON',
                url: 'json/ESP.geo.json'
            }
        },
        {
            name: 'Lisbon',
            group: 'Cities',
            source: {
                type: 'GeoJSON',
                url: 'json/LIS.geo.json'
            }
        } 
    ]
...
```
This makes more easy dynamic grouping because you can move layers between groups simply by changing/watching the `group` property on the layers. The hard part is that to avoid any problems the groups should be initialized first, before any other layer be pushed.

Don't know if it makes sense for you...

**Comment by lewisjared** [comment]
I like the group attribute on the layer, but it makes the group object redundant. The group attribute on the layer contains all the information required to create the groupings and gets around having to initialise the group layers first.

I have started a repo which converts ol3-layerswitcher to an angular module and makes some changes for use with this directive (https://github.com/lewisjared/angular-openlayers-layerswitcher). 

@tombatossals @juristr Do you have any comments

**Comment by claustres** [comment]
:+1:

**Comment by zhulinpinyu** [comment]
👍

**Comment by juristr** [comment]
Closing this as I merged it already and it is published in `v1.2.0`. Thx again for contributing

**Comment by dmishne** [comment]
hey, can you provided an example on how to use the grouping capability?
How can I control visibility of a group?
Thanks
